src/plugins/formulas/formulas.js

```diff
@@ -491,12 +492,13 @@ export class Formulas extends BasePlugin {
    * @private
    * @param {number} col Visual index of starter column.
    * @param {number} amount Amount of columns to be removed.
+   * @param {number[]} physicalColumns An array of physical columns removed from the data source.
    * @returns {*|boolean} If false is returned the action is canceled.
    */
-  onBeforeRemoveCol(col, amount) {
-    if (!this.engine.isItPossibleToRemoveColumns(this.sheetId, [col, amount])) {
-      return false;
-    }
+  onBeforeRemoveCol(col, amount, physicalColumns) {
+    return physicalColumns.every((physicalColumn) => {
```

Comment:
Keep in mind that the return value overwrites the first argument. When the engine allows removing columns or rows the `undefined` should be returned.

Reply:
Not `col`? Still, returning `true` should work just as well, right? https://github.com/handsontable/handsontable/blob/master/src/dataMap.js#L482

Reply:
No, if you try listen to the changes e.q 
```js
beforeRemoveCol(col) {
  console.log('beforeRemoveCol', col);
},
```
with Formulas enabled you've got `beforeRemoveCol true` in the console.

Reply:
True 👍